mlir/lib/Dialect/Tensor/Extensions/MeshShardingExtensions.cpp

```diff
@@ -22,10 +22,10 @@ using namespace mlir::mesh;
 
 namespace {
 
-// Sharding of tensor.empty
-struct EmptyOpShardingInterface
-    : public ShardingInterface::ExternalModel<EmptyOpShardingInterface,
-                                              tensor::EmptyOp> {
+// Sharding of tensor.empty/tensor.splat
+template<typename OpTy>
+struct CreatorOpShardingInterface
```

Comment:
Renaming a public interface may have unintended consequences. Here's hoping others have a proper look, as I don't know their downstream uses. @AntonLydike

Reply:
Not sure I understand your concern. This is a private class which implements an interface, but generically as a template. I can rename it to clarify that this is not an interface definition but an implementation.

Reply:
Ah, `namespace {`, not I see it. Ignore.